packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -214,10 +217,21 @@ export class ForkChoice implements IForkChoice {
           root: fromHexString(headRoot),
         });
       }
-
-      return (this.head = toBlockSummary(headNode));
+      prevHead = this.head;
+      return (this.head = newHead = toBlockSummary(headNode));
     } finally {
       if (timer) timer();
+      if (newHead && this.metrics) {
+        //if there is no prevHead, then isSameHead is false and isSameChain is true
+        const isSameHead = prevHead && toHexString(newHead.blockRoot) == toHexString(prevHead.blockRoot);
+        const isSameChain = !prevHead || isSameHead || this.isDescendant(prevHead.blockRoot, newHead.blockRoot);
+
+        if (isSameChain && !isSameHead) this.metrics.forkChoiceNewHeads.inc();
+        if (!isSameChain) this.metrics.forkChoiceNewChains.inc();
```

Comment:
The logic here is overlapping with https://github.com/ChainSafe/lodestar/blob/5c3331bebfbe233f053c4dcb19abc6e6e1c34394/packages/lodestar/src/chain/blocks/stateTransition.ts#L150

We may be able to track reorg events in lodestar package directly instead of here

Reply:
ok, let me check and figure out whats happening there.